sno/init.py

```diff
@@ -503,6 +520,82 @@ def build_meta_info(self):
         yield from gpkg.get_meta_info(db, layer=self.table)
 
 
+class ImportPostgreSQL(OgrImporter):
+    @classmethod
+    def postgres_url_to_ogr_conn_str(cls, url):
+        """
+        Takes a URL ('postgresql://..')
+        and turns it into a key/value connection string, prefixed by 'PG:' for OGR.
+
+        libpq actually handles URIs fine, but OGR doesn't :(
+        So to import via OGR we have to convert them.
+        https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING
+
+        We don't currently handle all of the things these URLs can contain.
+        In particular, we don't (currently) handle:
+         * domain sockets
```

Comment:
Is that because OGR doesn't handle them? They're the default on POSIX for local DBs, would be handy to support if it's straightforward.

Reply:
OGR handles them. 

It's mostly because I didn't want to take the time to understand the syntactical nuances, but I'll take a look at it tomorrow. I don't know if OGR supports them or not, I assume it does

Reply:
https://stackoverflow.com/a/29356068/2662 has syntax details (emphasis mine)

> The host component is interpreted as described for the parameter host. In particular, **a Unix-domain socket connection is chosen if the host part is either empty or starts with a slash**, otherwise a TCP/IP connection is initiated. Note, however, that the slash is a reserved character in the hierarchical part of the URI. So, to specify a non-standard Unix-domain socket directory, either omit the host specification in the URI and specify the host as a parameter, or percent-encode the path in the host component of the URI:
>
>     postgresql:///dbname?host=/var/lib/postgresql
>     postgresql://%2Fvar%2Flib%2Fpostgresql/dbname

Reply:
added support for those